ibis/common/tests/test_patterns.py

```diff
@@ -24,13 +24,13 @@
 
 from ibis.common.annotations import ValidationError
 from ibis.common.collections import FrozenDict
+from ibis.common.deferred import Call, var
```

Comment:
IIUC this PR isn't merely a refactor, but something that adds support for deferred expressions in the rewrite side of patterns. If so, could we add some tests for this? Apologies if I missed them, I didn't see them below in this file.

Reply:
The tests exercising the `Replace` pattern use these deferred objects. This functionality has been covered before this PR just without the rich syntax sugar provided by the deferred API.

Reply:
Sure. Since the "rich syntax sugar provided by the deferred API" is the new feature being added here though, I'd argue that there should be at least a test including a full example of this usage. Can you add at least a simple test showing how the pieces all fit together here?

Reply:
Added tests to `ibis/expr/operations/tests/test_rewrites.py`